### PR TITLE
Attach submission admin

### PR DESCRIPTION
**Pull request by tjroeder**
## Checklist
- [x] Tests written where required
- [x] All tests passing
- [ ] Unused code/prys/etc deleted
Testing Coverage at <percentage>%

## Changes Made
What (if any) features are you implementing?
- Add Helper modules for controllers, for exception handling, and json requests.
- Add Test Helper module for parsing json in test files.
- Add the creation of SubmissionAdmin when a Submission is made in SubmissionController#create.
- Add testing for SubmissionAdmin join table creation and edge case testing.

What (if anything) did you refactor?
- Refactored code to utilize the helper modules detailed above.

Were there any issues that arose?
- No issues at the moment.

Is there anything that you need from your teammate?
- If there are any additional edges cases that should be covered please let me know. As we will probably need to create a more robust set of edge case testing.

Any other comments, questions, or concerns?
- @BrianZanti I think one of our main thoughts for this PR was whether we should be doing the SubmissionAdmin creation in the Submissions controller as we are doing in this PR? Or should we have a redirect after creating the Submission to a new SubmissionAdmin controller and create it there with all the connections to each Admin. We believed it was advantageous to do it the way we have currently rather than the latter. One of the most prominent reasons was that it would open up access to another controller that we don't need. Any thoughts would be greatly appreciated. Thank you.

### CODE REVIEW

**Review by Sierra-T-9598 — APPROVED** [review]

File: app/controllers/concerns/response.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module Response
+  def json_response(object, status = :ok)
+    render json: object, status: status
+  end
+end
```
Comment (BrianZanti) [inline app/controllers/concerns/response.rb:5]: I like this!

File: app/controllers/concerns/exception_handler.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    end
+    
+    rescue_from ActiveRecord::RecordNotDestroyed do |e|
+      json_response({ errors: e.record.errors }, :unprocessable_entity)
+    end
+    
+    rescue_from ActionController::ParameterMissing do |e|
+      json_response({ errors: e.message }, :unprocessable_entity)
+    end
+  end
```
Comment (BrianZanti) [inline app/controllers/concerns/exception_handler.rb:20]: Looks good 👍 I think your choice of response codes are appropriate.

File: app/controllers/api/v1/submissions_controller.rb [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 
   def submission_params
     params.require(:submission).permit(:email, :profile, :video, :genre, :name, :song_title, :winner)
   end
+
+  def attach_submission_admin(submission)
+    Admin.all.each do |admin|
+      SubmissionAdmin.create!(admin_id: admin.id, submission_id: submission.id)
+    end
+  end
```
Comment (BrianZanti) [inline app/controllers/api/v1/submissions_controller.rb]: I would consider this data logic and recommend that you move it to the Submission model.

File: spec/support/request_spec_helper.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+module RequestSpecHelper
+  def json_parse(response)
+    JSON.parse(response, symbolize_names: true)
+  end
+end
```
Comment (BrianZanti) [inline spec/support/request_spec_helper.rb:5]: 👍

**Review by BrianZanti — CHANGES_REQUESTED** [review]
Overall, looking really good! I have one suggested change for you to look at.

File: app/controllers/api/v1/submissions_controller.rb (hunk 3 above)
Comment (tjroeder) [reply app/controllers/api/v1/submissions_controller.rb]: Ok sounds good, I will move the logic and re-push the change.

Comment (tjroeder) [reply app/controllers/api/v1/submissions_controller.rb]: @BrianZanti I have updated this logic and moved it to the model as a after_create callback action. I have also added testing for the new method. Please let me know if any additional changes are needed. Thanks!

**Review by arnaldoaparicio — APPROVED** [review]

**Review by Sierra-T-9598 — COMMENTED** [review]
Looks good to me!
